server/tests/src/test/java/org/infinispan/server/security/AuthenticationMultiRealmIT.java

```diff
@@ -0,0 +1,86 @@
+package org.infinispan.server.security;
+
+import static org.junit.Assert.assertEquals;
+
+import org.infinispan.client.hotrod.RemoteCache;
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.server.test.api.TestUser;
+import org.infinispan.server.test.core.LdapServerListener;
+import org.infinispan.server.test.core.category.Security;
+import org.infinispan.server.test.junit4.InfinispanServerRule;
+import org.infinispan.server.test.junit4.InfinispanServerRuleBuilder;
+import org.infinispan.server.test.junit4.InfinispanServerTestMethodRule;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+/**
+ * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
+ * @since 13.0
+ **/
+
+@Category(Security.class)
+public class AuthenticationMultiRealmIT {
+   @ClassRule
+   public static InfinispanServerRule SERVERS =
+         InfinispanServerRuleBuilder.config("configuration/AuthenticationMultiRealm.xml")
+               .addListener(new LdapServerListener())
+               .build();
+
+   @Rule
+   public InfinispanServerTestMethodRule SERVER_TEST = new InfinispanServerTestMethodRule(SERVERS);
+
+   public AuthenticationMultiRealmIT() {
+   }
+
+   @Test
+   public void testLDAP() {
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      SERVERS.getServerDriver().applyTrustStore(builder, "ca");
+      builder.security().authentication()
+            .saslMechanism("SCRAM-SHA-256")
+            .serverName("infinispan")
+            .realm("default")
+            .username(TestUser.ADMIN.getUser())
```

Comment:
`javax.security.sasl.SaslException: ELY05161: Unable to retrieve password for "admin"`